src/PrestaShopBundle/Twig/SearchFiltersExtension.php

```diff
@@ -0,0 +1,80 @@
+<?php
+/**
+ * 2007-2018 PrestaShop
+ *
+ * NOTICE OF LICENSE
+ *
+ * This source file is subject to the Open Software License (OSL 3.0)
+ * that is bundled with this package in the file LICENSE.txt.
+ * It is also available through the world-wide-web at this URL:
+ * https://opensource.org/licenses/OSL-3.0
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@prestashop.com so we can send you a copy immediately.
+ *
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to http://www.prestashop.com for more information.
+ *
+ * @author    PrestaShop SA <contact@prestashop.com>
+ * @copyright 2007-2018 PrestaShop SA
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ * International Registered Trademark & Property of PrestaShop SA
+ */
+namespace PrestaShopBundle\Twig;
+
+use Symfony\Component\HttpFoundation\RequestStack;
+use PrestaShop\PrestaShop\Core\Search\ControllerAction;
+
+/**
+ * Improve information from request, get the real action and controller's names.
+ */
+final class SearchFiltersExtension extends \Twig_Extension
+{
+    /**
+     * @var RequestStack
+     */
+    private $requestStack;
+
+    public function __construct(RequestStack $requestStack)
+    {
+        $this->requestStack = $requestStack;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function getFunctions()
+    {
+        return [
+            new \Twig_SimpleFunction('getController', [$this, 'getController']),
```

Comment:
Maybe a little overkill when you can do something like `app.request.attributes.get("_controller")`in twig

Reply:
read it again, you don't retrieve the same value at all ;)

When I can avoid to put business logic in template, I do it ^^

Reply:
> app.request The Request object that represents the current request
https://github.com/symfony/twig-bridge/blob/3513e775dfca0e25e8f908f3fc9a0a7aa3417c06/AppVariable.php#L102

This is exactly what twig-bridge does

Reply:
But btw I saw my mistake with getController and getAction ^^